queries/bash/highlights.scm

```diff
@@ -0,0 +1,110 @@
+[
+ "("
+ ")"
+ "{"
+ "}"
+ "["
+ "]"
+ ] @punctuation.bracket
+
+[
+ ";"
+ ";;"
+ (heredoc_start)
+ ] @punctuation.delimiter
+
+[
+ ">"
+ "<"
+ "&"
+ "&&"
+ "|"
+ "||"
+ "="
+ "=="
+ "!="
+ ] @operator
+
+[
+ (string) 
+ (raw_string)
+ (heredoc_body)
+] @string
+
+[
+ "if"
+ "then"
+ "else"
+ "elif"
+ "fi"
+ "case"
+ "in"
+ "esac"
+ ] @conditional
+
+[
+ "for"
+ "do"
+ "done"
+ "while"
+ ] @repeat
+
+[
+ "declare"
+ "export"
+ "local"
+ "readonly"
+ "unset"
+ ] @keyword
+
+[
+ (special_variable_name) 
+ ("$" (special_variable_name))
```

Comment:
That seems weird.. Doesn't they overlap ?

Reply:
Nope the parser is weird about some of the matches so they will look a bit "odd" here's an example with an without the second pair.
![before](https://user-images.githubusercontent.com/19380243/85404220-a33b9180-b52c-11ea-9a92-dc35d5886e28.png)
![after](https://user-images.githubusercontent.com/19380243/85404227-a59deb80-b52c-11ea-8f0f-706876ee39ff.png)

you can see here that in the first screenshot the "0" in the special variable is highlighted but not the "$" that goes with it. Surprisingly only the "$" get highlighted if you only use the pair.